### PR TITLE
Linear interpolation in angle

### PR DESCRIPTION
**Pull request by zigaosolin**
Fixed texture_progress to interpolate linearly in angle instead of "on rect".

### CODE REVIEW

File: scene/gui/texture_progress.cpp [hunk 1]
```diff
@@ -189,30 +191,30 @@ void TextureProgress::draw_nine_patch_stretched(const Ref<Texture> &p_texture, F
...
-				bottomright.y = last_section_size;
-			} break;
-			case FILL_BOTTOM_TO_TOP: {
-				src_rect.position.y += src_rect.size.y - width_texture;
-				src_rect.size.y = width_texture;
-				dst_rect.position.y += width_total - width_filled;
-				dst_rect.size.y = width_filled;
-				topleft.y = last_section_size;
-			} break;
+		case FILL_LEFT_TO_RIGHT: {
```
Comment (akien-mga) [inline scene/gui/texture_progress.cpp]: Why change the indentation? The new style you're using does not follow our guidelines: http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html

File: scene/gui/texture_progress.cpp [hunk 2]
```diff
@@ -224,90 +226,92 @@ void TextureProgress::_notification(int p_what) {
...
+				draw_nine_patch_stretched(under, FILL_LEFT_TO_RIGHT, 1.0);
+			}
+			if (progress.is_valid()) {
+				draw_nine_patch_stretched(progress, mode, get_as_ratio());
+			}
+			if (over.is_valid()) {
+				draw_nine_patch_stretched(over, FILL_LEFT_TO_RIGHT, 1.0);
+			}
+		}
+		else {
```
Comment (akien-mga) [inline scene/gui/texture_progress.cpp]: Same here, incorrect style.

Comment (zigaosolin) [reply scene/gui/texture_progress.cpp]: Thanks. Sorry, I did not have the time to update this. I have the tool for formatting now installed so it should not happen in the future (Visual Studio automatically changed the switch statement formatting).

**Comment by akien-mga** [comment]
This PR was superseded by #17422. I'll let you and @robfram check if the current state is good enough or if some changes made here would still be relevant.

**Pull request closed without merging**